src/solvers/exponentiate.jl

```diff
@@ -0,0 +1,42 @@
+function exponentiate_solver()
+    function solver(
+      init;
+      psi_ref!,
+      PH_ref!,
+      ishermitian=true,
+      issymmetric=true,
+      region,
+      sweep_regions,
+      sweep_step,
+      solver_krylovdim=30,
+      solver_maxiter=100,
+      solver_outputlevel=0,
+      solver_tol=1E-12,
+      substep,
+      normalize,
+      time_step,
+    )
+        #H=copy(PH_ref![])
+        H=PH_ref![] ###since we are not changing H we don't need the copy
+        # let's test whether given region and sweep regions we can find out what the previous and next region were
+        # this will be needed in subspace expansion
+        region_ind=sweep_step
+        next_region=region_ind==length(sweep_regions) ? nothing : first(sweep_regions[region_ind+1])
+        previous_region=region_ind==1 ? nothing : first(sweep_regions[region_ind-1])
```

Comment:
I realize this is a work in progress, but I'm a bit confused why this code logic is here since it isn't used by `exponentiate`.

Reply:
I guess I see from the comment above that this was just a proof of principle and will only be used in the subspace expansion function.

Reply:
Yes, this was just a placeholder to check whether the information required for subspace information can be passed to the solver. I'll remove it / move it to subspace expansion solver.

Reply:
Ok, let's discuss this later, this part of the design is hard to follow/keep track of so indicates to me that there is something too complicated about it.